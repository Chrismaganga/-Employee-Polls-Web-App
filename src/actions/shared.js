import { _getInitialData } from "../lib/_DATA";
import { receiveQuestions } from "./questions";
import { receiveUsers } from "./users";
import { setAuthedUser } from "./authedUser";
import { showLoading, hideLoading } from "react-redux-loading-bar";

export function handleInitialData() {
  return async (dispatch) => {
    dispatch(showLoading());
    const { users, questions } = await _getInitialData();
    dispatch(receiveUsers(users));
    dispatch(receiveQuestions(questions));
    dispatch(hideLoading());
  };
}

export function handleLogin(AUTHED_ID) {
  return async (dispatch) => {
    dispatch(setAuthedUser(AUTHED_ID));
  };
}

