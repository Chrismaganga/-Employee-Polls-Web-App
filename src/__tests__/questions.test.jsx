/* eslint-disable no-undef */

import { RECEIVE_QUESTIONS, ANSWER_QUESTION, SAVE_QUESTION } from '../actions/questions';
import questions from '../reducers/questions';

describe('questions reducer', () => {
  it('should match snapshot when RECEIVE_QUESTIONS action is dispatched', () => {
    const action = {
      type: RECEIVE_QUESTIONS,
      questions: {
        'q1': { id: 'q1', optionOne: { text: 'Option One', votes: [] }, optionTwo: { text: 'Option Two', votes: [] } },
        'q2': { id: 'q2', optionOne: { text: 'Option A', votes: [] }, optionTwo: { text: 'Option B', votes: [] } },
      },
    };

    const initialState = {};
    const newState = questions(initialState, action);

    expect(newState).toMatchSnapshot();
  });

  it('should match snapshot when ANSWER_QUESTION action is dispatched', () => {
    const action = {
      type: ANSWER_QUESTION,
      qid: 'q1',
      authedUser: 'user1',
      answer: 'optionOne',
    };

    const initialState = {
      'q1': {
        id: 'q1',
        optionOne: { text: 'Option One', votes: [] },
        optionTwo: { text: 'Option Two', votes: [] },
      },
    };

    const newState = questions(initialState, action);

    expect(newState).toMatchSnapshot();
  });

  it('should match snapshot when SAVE_QUESTION action is dispatched', () => {
    const action = {
      type: SAVE_QUESTION,
      question: {
        id: 'q3',
        optionOne: { text: 'New Option One', votes: [] },
        optionTwo: { text: 'New Option Two', votes: [] },
      },
    };

    const initialState = {};
    const newState = questions(initialState, action);

    expect(newState).toMatchSnapshot();
  });
});
