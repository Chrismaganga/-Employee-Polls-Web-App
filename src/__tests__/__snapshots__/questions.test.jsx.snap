// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`questions reducer should match snapshot when ANSWER_QUESTION action is dispatched 1`] = `
{
  "q1": {
    "id": "q1",
    "optionOne": {
      "text": "Option One",
      "votes": [
        "user1",
      ],
    },
    "optionTwo": {
      "text": "Option Two",
      "votes": [],
    },
  },
}
`;

exports[`questions reducer should match snapshot when RECEIVE_QUESTIONS action is dispatched 1`] = `
{
  "q1": {
    "id": "q1",
    "optionOne": {
      "text": "Option One",
      "votes": [],
    },
    "optionTwo": {
      "text": "Option Two",
      "votes": [],
    },
  },
  "q2": {
    "id": "q2",
    "optionOne": {
      "text": "Option A",
      "votes": [],
    },
    "optionTwo": {
      "text": "Option B",
      "votes": [],
    },
  },
}
`;

exports[`questions reducer should match snapshot when SAVE_QUESTION action is dispatched 1`] = `
{
  "q3": {
    "id": "q3",
    "optionOne": {
      "text": "New Option One",
      "votes": [],
    },
    "optionTwo": {
      "text": "New Option Two",
      "votes": [],
    },
  },
}
`;
